### PR TITLE
search address field implemented

### LINKED ISSUE #733: Input field for searching an address and jumping to it on map
* https://github.com/HSLdevcom/hsl-route-map-poc/blob/master/index.html#L304

https://api.digitransit.fi/geocoding/v1/search?text=pasila,helsinki

* show label as a search result

http://kartat.hsl.fi/linjakartta/

### PR DESCRIPTION
**Pull request by Divinare**
Closes #733

* Added TODO to move createDivIcon to /utils. Wasn't sure because maybe it doesn't belong there and can as well be where it is (or could be moved to some other place)
* Had to modify ApiClient a bit so that we could send request without 'include' param

### CODE REVIEW

File: src/components/map/AddressSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            this.setState({
+                input: newValue,
+            });
+            this.requestAddress(newValue);
+        } else {
+            this.unselectSearchResult();
+        }
+    }
+    private requestAddress = async (value: string) => {
+        const GEOCODER_ADDRESS = 'https://api.digitransit.fi/geocoding/v1/search';
```
Comment (kronken) [inline src/components/map/AddressSearch.tsx]: Address in constants

File: src/util/ApiClient.ts [hunk 2]
```diff
@@ -46,15 +48,26 @@ class ApiClient {
...
     private sendRequest = async (method: RequestMethod, endpoint: endpoints, object: any) => {
+        const entityUrl = this.getEntityUrl(endpoint);
+        return this.sendRequestToUrl(method, entityUrl, object, 'include');
+    }
+
+    private getEntityUrl = (entityName: endpoints) => {
+        return `${API_URL}/${entityName}`;
+    }
+
+    public sendRequestToUrl = async (
```
Comment (kronken) [inline src/util/ApiClient.ts]: the difference in name between sendRequest and sendRequestToUrl is not declarative enough.

Comment (kronken) [reply src/util/ApiClient.ts]: I would suggest:
sendRequestToUrl => sendRequest
sendRequest => sendJoreMapRequest

**Comment by kronken** [comment]
Found bug:
Cannot click on first search result

File: src/util/ApiClient.ts (hunk 2 above)
Comment (Divinare) [reply src/util/ApiClient.ts]: Fixed, its much better now

**Pull request merged**